### ISSUE TITLE
ENVIRONMENT=web, optimised builds, Closure and stdio

### ISSUE THREAD

**Issue by cwoffenden** [emscripten-core/emscripten#8010]
We're seeing an issue which I've narrowed down to Closure use on optimised builds with stdio. I'm guesing it's related to #7933 and #7803, but here's a standalone reproducible case nonetheless:
```
#include <cstdio>
int main() {
	fopen("test", "wb");
	return 0;
}
```
Build with:
```
emcc -s ENVIRONMENT=web -s STRICT -Os --closure 1 test.cpp
```
It will fail with Closure's dump followed by `4 error(s), 2 warning(s)`. Remove the `ENVIRONMENT` flag or set it to `node`, or remove the `-O` option, and it's fine. This was fine in 1.38.24 but fails after.

**Comment by juj** [comment]
Thanks, good test case. Fix PR up in #8010.

**Comment by juj** [comment]
There may be more items like this, the general pattern is that after 1.38.25, we started to be more strict about ENVIRONMENT=web builds for Closure, and we no longer feed Closure information about node.js extern symbols if one is doing a ENVIRONMENT=web build, so the generated code should not have any Node.jsisms outputted in it. Though it is hard to go through all code to verify, so regressions are possible. Fortunately these are easy to fix once identified, so if you find any more issues, test cases like this are greatly appreciated!
